### PR TITLE
feat(styled-system): create `gradients` theme key

### PR DESCRIPTION
**Pull request by TylerAPfledderer**
Add the `gradient` theme key to be able to create tokens for gradient styles.

> Applies to the `bgGradient`, `bgImg`, `bgImage`, and `backgroundImage` style props.

Due to the current setup under the hood of the [Background Gradient API](https://chakra-ui.com/docs/styled-system/gradient#background-gradient-api), I do not believe it is worth the investment to make the logic work in a theme config. There are probably easier and cheaper ways to acquire the color theme tokens;

- CSS Variables explicitly: 
```
lightLinearGradient: 'linear-gradient(to right, var(--chakra-colors-gray-200), var(--chakra-colors-blue-500))'
```
- Importing the colors object and use interpolation: 
```ts
lightLinearGradient: `linear-gradient(to right, ${colors.gray.200}, ${colors.blue.500})`
```

### CODE REVIEW

File: packages/core/styled-system/src/config/background.ts
Comment (segunadebayo) [inline packages/core/styled-system/src/config/background.ts:1]: Let's connect the gradients token to `backgroundImage`, `bgImage` and `bgGradient`

**Comment by TylerAPfledderer** [comment]
@segunadebayo made the updates as requested.

File: .changeset/sixty-otters-pretend.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+// SomeComponent.tsx
+
+<Box bgGradient='lightBgGradient' />
+```
+
+🚨 NOTE: The
+[Background Gradient API](https://chakra-ui.com/docs/styled-system/gradient#background-gradient-api)
+can not be used in a token as the conversion is done when the api is used
+directly on a prop and not when compiling the theme config
```
Comment (TylerAPfledderer) [inline .changeset/sixty-otters-pretend.md:27]: @segunadebayo did I explain this correctly?

File: packages/core/styled-system/src/config/background.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { t, Token, transforms } from "../utils"
 export const background: Config = {
   background: t.colors("background"),
   backgroundColor: t.colors("backgroundColor"),
-  backgroundImage: t.propT("backgroundImage", transforms.bgImage),
```
Comment (segunadebayo) [inline packages/core/styled-system/src/config/background.ts:8]: This one is a breaking change. I was hoping we could add this while preserving the existing transforms for `bgGradient`

Comment (TylerAPfledderer) [reply packages/core/styled-system/src/config/background.ts:8]: @segunadebayo I ended up moving it's logic over to the `parseGradient` function, hence the tests I added still passing. Will this still mot work?

Comment (TylerAPfledderer) [reply packages/core/styled-system/src/config/background.ts:8]: @segunadebayo I moved it's logic over to the `parseGradient` function, hence the test I added passed with the `url()`. Is there a check I am missing?

Comment (TylerAPfledderer) [reply packages/core/styled-system/src/config/background.ts:8]: I'm just now recognizing that I should be adding tests to `parse-gradient.test.ts` instead. But it appears there are no tests failing there at the moment.

Comment (segunadebayo) [reply packages/core/styled-system/src/config/background.ts:8]: Got it. I'm going to double check that.

**Pull request merged**